lib/workload/stateless/universal-event-archiver/function/achiver.py

```diff
@@ -0,0 +1,51 @@
+import json
+import boto3
+from datetime import datetime, timezone
+import os
+import re
+import logging
+
+# Initialize S3 client
+s3 = boto3.client('s3')
+
+# The name of the bucket
+BUCKET_NAME = os.getenv('BUCKET_NAME')
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def handler(event, context):
+    
+    # Current timestamp
+    now = datetime.now(timezone.utc)
+    time_stamp = str(now.timestamp()) # for object name
+    time_stamp_details = now.strftime("%Y-%m-%d__%H-%M-%S") # for tagging
+    
+    # Extract the event title (type) from detail type
+    event_title = sanitize_string(event.get('detail-type', 'undefinedEvent'))
+    
+    # Formatting the S3 key with year/month/day partitioning
+    key = f'events/{now.year}/{now.month:02}/{now.day:02}/{event_title+'_'+time_stamp}.json'
```

Comment:
Thought: If we aim to datalake down the track and, let it crawl by glue then the s3 key path should be more like key-value partition pair?

```
s3://{bucket_name}/events/year=2024/month=04/day=16/WorkflowRequest_1713252338.243297.json
```

Reply:
Hm... 
Perhaps one for @brainstorm , not sure how much difference that makes.
I was following how CloudTrail structures their log output (that's just `yyyy/mm/dd/`), but happy with what works best.

Reply:
yes, I will add the key-vaule pair if needed. just a little change, some research on key-value pair
> Benefits of Using Key-Value Partition Pairs:
> Query Optimization: Services like Amazon Athena can skip over entire chunks of data during a query if the partition keys are irrelevant to the query, significantly improving performance.
> Cost Efficiency: By structuring data into partitions, you can reduce the amount of data scanned by queries, thus reducing the costs associated with data scanning operations.
> Easier Data Management: Having a clear partitioning scheme makes it easier to manage and navigate large datasets. For example, you can set up lifecycle policies that are specific to certain partitions (e.g., archiving last year's data).

Reply:
OK, then let's go with the key-value notation.